### PR TITLE
Submit map button

### LINKED ISSUE #1170: Add submit button to maps
image

Wireframes here:

https://www.figma.com/file/FYobD6UsJZoNxlpyxUOlR5/Application?node-id=0%3A2735

AC:
 - New field(s) added to `ProjectTemplate` to indicate it is for a contest, contest text (like other template fields, this will only be configurable via SQL for now)
 - Submit button shown for any project created from a template with a contest
 - Content of Next Steps modal should come from new `ProjectTemplate` field, should allow for HTML for linking to external sites
 - Submitted maps should be read-only

### PR DESCRIPTION
**Pull request by maurizi**
## Overview

Adds workflow for submitting a map for a contest.

Two new fields have been added to `ProjectTemplate`:
 - `contestActive`: controls whether the "Submit" button is shown for maps connected to this template
 - `contestNextSteps`: HTML to display in the Next steps modal displayed after submission (optional, modal not shown if not set)
 
 Submission date is tracked on a new `Project.submittedDt` field, which also causes the submitted map to become read-only.

### Checklist

- [x] Description of PR is in an appropriate section of `CHANGELOG.md` and grouped with similar changes, if possible

### Demo

Submitting a map w/o next steps HTML set:
submit-no-next-steps

Submitting a map with next steps HTML set:
submit-next-steps

## Testing Instructions

- Use `scripts/load-dev-data` to reload your test data (it should make a contest active on the PA House of Representatives template for the Azavea organization), or use `scripts/dbshell` to set up contest fields for a project template of your choice.
- Create a map for that template
  - There should be a "Submit" button in the upper-right. Submitting your map should display a success toast and open the Next steps modal. Clicking the "Submitted" button should display a link to re-open the next steps modal.
  - Using `scripts/dbshell`, update your project template to set the next steps HTML to be empty: `UPDATE project_template SET contest_next_steps = '';`
  - Submit another map - you should still see the toast, but not the Next steps modal, and there should no longer be a link to it.
  - Using `scripts/dbshell`, update your project template to set the contest to be inactive: `UPDATE project_template SET contest_active = FALSE;`. You should still see the "Submitted" button for maps that were already submitted, but should no longer see a "Submit" button for maps that were not yet submitted.
- A few minutes after submitting a map, go to the Evaluate > Competitiveness detail. The "View on PlanScore" link should be present, not the "Upload to PlanScore" button.

Closes #1170

### CODE REVIEW

File: src/client/components/ProjectHeader.tsx [hunk 1]
```diff
@@ -53,14 +54,15 @@ const HeaderDivider = () => {
...
+  readonly isOwnProject: boolean;
 }
 
 const EvaluateButton = ({ evaluateMode }: { readonly evaluateMode: boolean }) => (
   <Box sx={{ position: "relative" }}>
     <Button
       sx={{
         ...{
-          variant: "buttons.primary",
+          variant: "buttons.secondary",
```
Comment (maurizi) [inline src/client/components/ProjectHeader.tsx:65]: @jfrankl do we want to switch the "Evaluate" button to _always_ use secondary colors, or should it get the primary blue color when the submit button isn't visible like it does now?

File: src/client/components/map/SubmitMapButton.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        <Box>
+          <Heading as="h3">Submit map</Heading>
+          {project?.submittedDt ? (
+            <React.Fragment>
+              Your map was submitted to {organization.name} at{" "}
+              {project.submittedDt.toLocaleTimeString(undefined, { timeStyle: "short" })} on{" "}
+              {project.submittedDt.toLocaleDateString(undefined, { dateStyle: "long" })}.
+              {project?.projectTemplate?.contestNextSteps ? (
+                <React.Fragment>
+                  View{" "}
```
Comment (KlaasH) [inline src/client/components/map/SubmitMapButton.tsx:56]: These two sentences are getting smashed together: 
image

File: src/client/components/map/SubmitMapButton.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                  <Themed.a href="void:0" onClick={() => store.dispatch(showSubmitMapModal(true))}>
+                    next steps
+                  </Themed.a>
```
Comment (KlaasH) [inline src/client/components/map/SubmitMapButton.tsx:59]: When I click this link, it shows me this:
image

If I click the "Open xdg-open" button, it launches a new window in my default browser (which is not the one I was viewing the site in) with "void:0" in the address bar.

**Review by KlaasH — CHANGES_REQUESTED** [review]
The code looks good, and the testing steps all went as expected.  A couple small notes below (two capitalization/spacing notes and one weird warning window), and I also noticed one issue after I submitted: the "Submit" text says the map won't be editable after it's submitted, and the overlay in the corner of the map shows up saying it's read-only, but I can still edit it:
image

If I reload the tab, the editing interface is gone, but submitting without reloading doesn't seem to be fully switching things over to read-only.

**Review by rachelekm — COMMENTED** [review]
This worked well for me and I was able to successfully submit and see the message/modal/html where appropriate. My maps were read-only after submitting, however I did find that I could submit while still editing a district. In that instance, it was still editable after submit until I hit "accept" for the district and then it went back to being a non-editable, read-only map. 
Screen Shot 2022-03-29 at 6 31 49 PM

**Comment by maurizi** [comment]
@KlaasH thanks for the review! Somehow I hadn't even noticed the map was still interactable - my eyes just saw the tool selection buttons go away and assumed I was good 😳.

All issues should be addressed now.

**Comment by jfrankl** [comment]
@maurizi looks good! One request: could you please reduce the padding on the bottom of this panel so it closely matches the padding at the top of the padding?

image

**Review by KlaasH — APPROVED** [review]
:+1: Looks good!

**Pull request merged**